src/h5web/toolbar/controls/DomainSlider/DomainSlider.test.tsx

```diff
@@ -0,0 +1,336 @@
+  const maxThumb = screen.getByRole('slider', { name: /max/u });
+  const minInput = screen.getByLabelText('min');
+  const maxInput = screen.getByLabelText('max');
+
+  userEvent.type(maxInput, '0000{enter}');
+  expect(maxInput).toHaveValue('8.98847e+307');
+  expect(maxThumb).toHaveAttribute('aria-valuenow', '100');
+
+  userEvent.type(minInput, '0000{enter}');
+  expect(minInput).toHaveValue('−8.98847e+307');
+  expect(minThumb).toHaveAttribute('aria-valuenow', '1');
+
+  maxThumb.focus();
+  pressKey('ArrowLeft');
+  expect(maxInput).toHaveValue('5.40006e+301');
+  expect(maxThumb).toHaveAttribute('aria-valuenow', '99'); // does not jumb back to 81
+
+  // Remove focus now to avoid state update after unmount
+  maxThumb.blur();
+});
+
+test('show min/max autoscale toggles in tooltip (pressed by default)', async () => {
+  renderApp();
+
+  const editBtn = await screen.findByRole('button', { name: 'Edit domain' });
+  userEvent.hover(editBtn);
+
+  expect(screen.getByText('Autoscale')).toBeVisible();
+
+  const minBtn = screen.getByRole('button', { name: 'Min' });
+  const maxBtn = screen.getByRole('button', { name: 'Max' });
+  expect(minBtn).toHaveAttribute('aria-pressed', 'true');
+  expect(maxBtn).toHaveAttribute('aria-pressed', 'true');
+});
+
+test('control min/max autoscale behaviour', async () => {
+  renderApp();
+
+  const minThumb = await screen.findByRole('slider', { name: /min/u });
+  userEvent.hover(minThumb);
+
+  const minBtn = screen.getByRole('button', { name: 'Min' });
+  const maxBtn = screen.getByRole('button', { name: 'Max' });
+  const maxInput = screen.getByLabelText('max');
+
+  // Moving min thumb disables min autoscale
+  minThumb.focus();
+  pressKey('ArrowRight');
+  expect(minBtn).toHaveAttribute('aria-pressed', 'false');
+  expect(maxBtn).toHaveAttribute('aria-pressed', 'true'); // unaffected
+
+  // Editing max disables max autoscale
+  userEvent.type(maxInput, '0{enter}');
+  expect(maxInput).toHaveValue('4e+20');
+  expect(maxBtn).toHaveAttribute('aria-pressed', 'false');
+
+  // Re-enable max autoscale
+  userEvent.click(maxBtn);
+  expect(maxInput).toHaveValue('4e+2');
+  expect(maxBtn).toHaveAttribute('aria-pressed', 'true');
+  expect(minBtn).toHaveAttribute('aria-pressed', 'false'); // unaffected
+});
+
+test('handle empty domain', async () => {
+  renderApp();
+
+  const visArea = await screen.findByRole('figure');
+  const editBtn = screen.getByRole('button', { name: 'Edit domain' });
+  userEvent.click(editBtn);
+
+  const minInput = screen.getByLabelText('min');
+  const maxInput = screen.getByLabelText('max');
+  const minThumb = screen.getByRole('slider', { name: /min/u });
+  const maxThumb = screen.getByRole('slider', { name: /max/u });
+
+  // Give min the same value as max
+  userEvent.clear(minInput);
+  userEvent.type(minInput, '400{enter}');
+  expect(minThumb).toHaveAttribute('aria-valuenow', '58'); // not quite in the middle because of symlog
```

Comment:
It would definitely not simplify the test, since I'd have to manually switch the scale to linear by interacting with the scale selector. In a way, I feel like the fact that it's 58 actually demonstrates that the slider works as expected in regard to scales!